core/src/consensus/tendermint/backup.rs

```diff
@@ -78,14 +167,29 @@ pub fn restore(db: &dyn KeyValueDB) -> Option<BackupData> {
     })
 }
 
-fn find_proposal(votes: &[ConsensusMessage], height: Height, view: View) -> Option<H256> {
-    votes
-        .iter()
-        .rev()
-        .map(|vote| &vote.on)
-        .find(|vote_on| {
-            vote_on.step.step == Step::Propose && vote_on.step.view == view && vote_on.step.height == height
-        })
-        .map(|vote_on| vote_on.block_hash)
-        .unwrap_or(None)
+fn load_v1(db: &dyn KeyValueDB) -> Option<BackupDataV1> {
+    #[derive(RlpDecodable)]
+    struct Backup {
+        height: Height,
+        view: View,
+        step: Step,
+        votes: Vec<ConsensusMessage>,
+        finalized_view_of_previous_block: View,
+        finalized_view_of_current_block: Option<View>,
+    }
+
+    let value = db.get(db::COL_EXTRA, BACKUP_KEY).expect("Low level database error. Some issue with disk?")?;
+    let backup: Backup = rlp::decode(&value);
```

Comment:
Would you check that it is possible to `Backup::decode(&value)`?

Reply:
That is not working. `RlpDecodable`'s `decode`is used in the internal code. `rlp::decode` is a highlevel function. In this case `rlp::decode` is better.